### ISSUE TITLE
Request: Support thesaurus.com

### ISSUE THREAD

**Issue by idbrii** [Ron89/thesaurus_query.vim#31]
Great plugin!

Looks like this would help support the new thesaurus.com output format:

https://github.com/Manwholikespie/thesaurus

Not sure if you'd want to include their python module directly?

**Comment by Ron89** [comment]
Yes, I have seen that plugin and was thinking the same. Will try to port it in as soon as I'm free.

**Comment by Ron89** [comment]
Hi, @idbrii , I have improved the usability of this feature and now merged it into main branch (commit 5de8051). Please try it out and give me your feedback. If you think it satisfies your need, please close this issue. Thanks.

**Comment by Ron89** [comment]
@Manwholikespie , Hi, it's nice to see you here~ Your Python project works great. It's just that this is a first time I added Python dependency for this plugin, and I used to use `ModuleNotFoundError` to detect the existence of the required dependency. That caused some problem on my Mac later on (for some reason, it doesn't have `ModuleNotFoundError` in the `ImportError`). So I changed into a more Pythonic way to find out the dependency and prompt user to install it. The result is satisfying already. I'd love to be able to get rid of the dependency prompt completely with some kind of auto-install mechanism. But Vim/NeoVim disallowed the embedded Python to spawn child process. So I guess there is no where around this.

I didn't think much about moneytising it though. I guess looking at people enjoying using my software is satisfying enough for me (inner demons, haha). I do feel complicated when dealing with thesaurus.com though. I do like the quality of the thesaurus resource it has. But I don't know why it just have to act against open access... German's equivalent, Open Thesaurus `https://www.openthesaurus.de/`is so much more friendlier in this regard. It's great that your work exists. :)
